Tests/OAuth/ResourceOwner/Auth0ResourceOwnerTest.php

```diff
@@ -74,4 +130,60 @@ protected function setUpResourceOwner($name, HttpUtils $httpUtils, array $option
 
         return parent::setUpResourceOwner($name, $httpUtils, $options);
     }
+
+    /**
+     * Mocks the {@see HttpClient::sendRequest} method with assertions on the request.
+     *
+     * @param string $expectedRequestUri
+     * @param string $expectedRequestMethod
+     * @param array  $expectedAuthorizationHeader
+     * @param array  $expectedAuth0ClientRequestHeader
+     * @param string $expectedRequestBodyContents
+     */
+    private function mockHttpClientSendRequestWithRequestAssertions(
+        string $expectedRequestUri,
+        string $expectedRequestMethod,
+        array $expectedAuthorizationHeader,
+        array $expectedAuth0ClientRequestHeader,
+        string $expectedRequestBodyContents
+    ): void {
+        $this->httpClient->expects($this->once())
+            ->method('sendRequest')
+            ->with(
+                $this->callback(function (RequestInterface $request) use ($expectedRequestUri, $expectedRequestMethod, $expectedAuthorizationHeader, $expectedAuth0ClientRequestHeader, $expectedRequestBodyContents) {
+                    $this->assertEquals($expectedRequestUri, $request->getUri());
+                    $this->assertSame(
+                        $expectedRequestMethod,
+                        $request->getMethod(),
+                        'The request should be send as POST request.'
```

Comment:
I actually forgot to adjust the description of the assertion. It's also used in the `getUserInformation` test (a GET request). I'll provide a follow-up PR for this.